storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -657,13 +660,24 @@ protected void testShouldRecordFinalizedBlocksAndStates(
         break;
       case PRUNE:
         // Check pruned states
-        final List<Bytes32> unavailableRoots =
-            allBlocksAndStates.stream().map(SignedBlockAndState::getRoot).collect(toList());
+        final List<UnsignedLong> unavailableRoots =
+            allBlocksAndStates.stream().map(SignedBlockAndState::getSlot).collect(toList());
         assertStatesUnavailable(unavailableRoots);
         break;
     }
   }
 
+  private void assertFinalizedBlocksAvailableViaStream(final ChainBuilder primaryChain) {
+    try (final Stream<SignedBeaconBlock> stream =
+        database.streamFinalizedBlocks(UnsignedLong.valueOf(1), UnsignedLong.valueOf(3))) {
+      assertThat(stream)
+          .containsExactly(
+              primaryChain.getBlockAtSlot(1),
+              primaryChain.getBlockAtSlot(2),
+              primaryChain.getBlockAtSlot(3));
```

Comment:
Shouldn't block `7` also be available?  It was finalized.

Reply:
We're specifically requesting blocks 1-3 to test the range is respected.  I've made it a more generic function which hopefully make that clearer as well.